pkg/helios/cas/cas.go

```diff
@@ -0,0 +1,398 @@
+	go store.backgroundWriter()
+	
+	// Start error handler goroutine
+	go store.errorHandler()
+	
+	return store, nil
+}
+
+// backgroundWriter handles async disk writes for maximum performance
+func (s *BLAKE3Store) backgroundWriter() {
+	for writeOp := range s.writeQueue {
+		if err := os.WriteFile(writeOp.filePath, writeOp.content, 0644); err != nil {
+			// Send error to error handler for logging/retry
+			select {
+			case s.errorQueue <- fmt.Errorf("background write failed for %s: %w", writeOp.filePath, err):
+			default:
+				// Error queue full, error will be dropped but write continues
+			}
+		}
+		s.wg.Done()
+	}
+}
+
+// errorHandler processes background write errors
+func (s *BLAKE3Store) errorHandler() {
+	for err := range s.errorQueue {
+		// Log error to stderr for proper error handling
+		// Using fmt.Fprintf to stderr instead of stdout for errors
+		fmt.Fprintf(os.Stderr, "[ERROR] BLAKE3Store background write failed: %v\n", err)
+		
+		// TODO: In production, this should:
+		// 1. Log to proper logging system (slog, logrus, zap)
+		// 2. Implement retry logic with exponential backoff
+		// 3. Send metrics/alerts for monitoring
+		// 4. Consider circuit breaker pattern for persistent failures
+	}
+}
+
+// EnableMemoryMode switches to memory-only mode for maximum performance (<70μs targets)
+func (s *BLAKE3Store) EnableMemoryMode() {
+	s.memoryMode = true
+}
+
+// Store saves content and returns its BLAKE3 hash
+func (s *BLAKE3Store) Store(content []byte) (types.Hash, error) {
+	// Quick check if store is closed (avoid lock for performance)
+	if s.closed {
+		return types.Hash{}, fmt.Errorf("store is closed")
+	}
+	// Get hasher from pool for zero allocation
+	hasher := s.hasherPool.Get().(*blake3.Hasher)
+	defer func() {
+		hasher.Reset() // Reset for reuse
+		s.hasherPool.Put(hasher)
+	}()
+
+	// Compute BLAKE3 hash using pooled hasher
+	hasher.Write(content)
+	digest := hasher.Sum(nil)
+
+	hash := types.Hash{
+		Algorithm: types.BLAKE3,
+		Digest:    digest,
+	}
+
+	// Pre-compute hash key once (use hex digest as key)
+	hashKey := fmt.Sprintf("%x", digest)
+	
+	// Check if already exists to avoid duplicate work
+	s.mutex.RLock()
+	if _, exists := s.cache[hashKey]; exists {
+		s.mutex.RUnlock()
+		return hash, nil // Content already stored
+	}
+	s.mutex.RUnlock()
+
+	// Store in cache for fast access (zero-copy when possible)
+	s.mutex.Lock()
+	s.cache[hashKey] = make([]byte, len(content))
+	copy(s.cache[hashKey], content)
+	s.mutex.Unlock()
+	
+	// Cache the hex key for future lookups
+	s.keyMutex.Lock()
+	s.keyCache[string(digest)] = hashKey
+	s.keyMutex.Unlock()
+
+	// Store persistently to disk - async for performance or skip in memory mode
+	if !s.memoryMode {
+		filePath := filepath.Join(s.storePath, hashKey)
+		
+		// Critical: Add to WaitGroup BEFORE attempting to send to channel
+		// to prevent race condition where Done() is called before Add()
+		s.wg.Add(1)
+		
+		// Use async writes for better performance
```

Comment:
Please gate this send with a done channel to avoid send-after-close during shutdown (see Close()). Pattern:\n\nselect {\ncase s.writeQueue <- op:\ncase <-s.done:\n    s.wg.Done()\n    // fallback to sync write or return error\n}\n\nIn Close(), close(done) first so no further sends happen before closing channels.